File: src/api/panel/searchEngine.ts
import request from '@/utils/request'

const api = {
    getList: '/panel/searchEngine/getList',
    add: '/panel/searchEngine/add',
    update: '/panel/searchEngine/update',
    delete: '/panel/searchEngine/delete',
    updateSort: '/panel/searchEngine/updateSort',
}

export const getList = () => {
    return request({
        url: api.getList,
        method: 'post',
    })
}

export const add = (data: { title: string; url: string; iconSrc: string }) => {
    return request({
        url: api.add,
        method: 'post',
        data,
    })
}

export const update = (data: { id: number; title: string; url: string; iconSrc: string; sort?: number }) => {
    return request({
        url: api.update,
        method: 'post',
        data,
    })
}

export const deletes = (data: { id: number }) => {
    return request({
        url: api.delete,
        method: 'post',
        data,
    })
}

export const updateSort = (data: { items: { id: number; sort: number }[] }) => {
    return request({
        url: api.updateSort,
        method: 'post',
        data,
    })
}

File: service/router/panel/searchEngine.go
package panel

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

// InitSearchEngine 初始化搜索引擎相关路由
func InitSearchEngine(router *gin.RouterGroup) {
	searchEngineApi := api_v1.ApiGroupApp.ApiPanel.SearchEngine

	// 需要登录的路由
	private := router.Group("", middleware.LoginInterceptor)
	{
		private.POST("/panel/searchEngine/add", searchEngineApi.Add)
		private.POST("/panel/searchEngine/update", searchEngineApi.Update)
		private.POST("/panel/searchEngine/delete", searchEngineApi.Delete)
		private.POST("/panel/searchEngine/updateSort", searchEngineApi.UpdateSort)
	}

	// 公开模式下可以获取搜索引擎列表
	public := router.Group("", middleware.PublicModeInterceptor)
	{
		public.POST("/panel/searchEngine/getList", searchEngineApi.GetList)
	}
}
